partiql-test-data/partiql_test_schema.isl

```diff
@@ -0,0 +1,103 @@
+schema_header::{
+}
+
+type::{
+    name: PartiQLTestDocument,
+    type: document,
+    element: { one_of: [TestCase, Namespace] }
+}
+
+type::{
+    name: Namespace,
+    type: list,
+    element: { one_of: [TestCase, Namespace] }
+}
+
+type::{
+    name: TestCase,
+    type: { one_of: [PassParseTestCase, FailParseTestCase, FailSemanticTestCase] }
+}
+
+type::{
+    name: PassParseTestCase,
+    type: struct,
+    annotations: required::[parse],
+    fields: {
+        name: { type: string, occurs: required },
+        statement: { type: string, occurs: required },
+        assert: {
+            type: {
+                one_of: [
+                    ParseOkAssertion,
+                    { type: list, contains: [{result: ParseOk}] }
+                ]
+            },
+            occurs: required
+        }
+    }
+}
+
+type::{
+    name: FailParseTestCase,
+    type: struct,
+    annotations: required::[parse],
+    fields: {
+        name: { type: string, occurs: required },
+        statement: { type: string, occurs: required },
+        assert: {
+            type: {
+                one_of: [
+                    ParseErrorAssertion,
+                    { type: list, contains: [{result: ParseError}] }
+                ]
+            },
+            occurs: required
+        }
+    }
+}
+
+type::{
+    name: FailSemanticTestCase,
+    type: struct,
+    annotations: required::[semantic],
+    fields: {
+        name: { type: string, occurs: required },
+        statement: { type: string, occurs: required },
+        assert: {
+            type: {
+                one_of: [
+                    SemanticErrorAssertion,
+                    { type: list, contains: [{result: SemanticError}] }
+                ]
+            },
+            occurs: required
+        }
+    }
+}
+
+type::{
+    name: ParseOkAssertion,
+    type: struct,
+    fields: {
+        result: { type: symbol, valid_values: [ParseOk], occurs: required }
+    }
+}
+
+type::{
+    name: ParseErrorAssertion,
+    type: struct,
+    fields: {
+        result: { type: symbol, valid_values: [ParseError], occurs: required }
+    }
+}
+
+type::{
+    name: SemanticErrorAssertion,
+    type: struct,
+    fields: {
+        result: { type: symbol, valid_values: [SemanticError], occurs: required }
+    }
+}
```

Comment:
Maybe we can abstract assertions further? For example, we could have a `AssertPass` and `AssertFail` assertions and then using those directly in the test cases; they could still use `valid_values` for `result`—e.g. `ParseOk`, etc.)

Reply:
I think having concretely different assertions is probably good, as they mean different things even if they happen to have some similar fields. The assertions are the different 'strategies' to apply to the test case.

The test cases on the other hand, largely seem like a container for the inputs and the 'strategies' to run against them.

Reply:
Maybe if I explain my point of view it makes more sense: I think as a test suite we need to provide means for defining test data, tests, and assertions. For assertions I consider them as functions that their execution either leads to throwing an exception or not. Following this mental model, I was thinking along the lines of building our assertions data-model as mentioned.